src/main/java/gregtech/api/recipes/category/GTRecipeCategory.java

```diff
@@ -0,0 +1,104 @@
+package gregtech.api.recipes.category;
+
+import gregtech.api.recipes.RecipeMap;
+import it.unimi.dsi.fastutil.objects.Object2ObjectOpenHashMap;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.Map;
+
+public final class GTRecipeCategory {
+
+    private static final Map<String, GTRecipeCategory> categories = new Object2ObjectOpenHashMap<>();
+
+    private final String modid;
+    private final String name;
+    private final String uniqueID;
+    private final String unlocalizedName;
+    private final RecipeMap<?> recipeMap;
+    private Object icon;
+
+    /**
+     * Create a GTRecipeCategory
+     *
+     * @param modid the mod id of the category
+     * @param name the name of the category, used in translation. Uses the recipeMap's translation key if null.
+     * @param recipeMap the recipemap that accepts this category
+     * @return the new category
+     */
+    @Nonnull
+    public static GTRecipeCategory create(@Nonnull String modid, @Nullable String name, @Nonnull RecipeMap<?> recipeMap) {
+        String key = name == null ? recipeMap.getUnlocalizedName() : name;
+        return categories.computeIfAbsent(key, (k) -> new GTRecipeCategory(modid, name, recipeMap));
+    }
+
+    private GTRecipeCategory(@Nonnull String modid, @Nullable String name, @Nonnull RecipeMap<?> recipeMap) {
+        this.modid = modid;
+        this.name = name == null ? recipeMap.getUnlocalizedName() : name;
+        this.uniqueID = modid + ':' + this.name;
```

Comment:
You are basically constructing a ResourceLocation here with the string concatenation, would it be better to just use a RL?

Reply:
JEI uses the raw string form, and we have no other use for a ResourceLocation for categories.